b2sdk/transfer/outbound/upload_manager.py

```diff
@@ -39,12 +40,25 @@ class UploadManager(TransferManager, ThreadPoolMixin):
     Handle complex actions around uploads to free raw_api from that responsibility.
     """
 
-    MAX_UPLOAD_ATTEMPTS = 5
+    # default retry time, in seconds, when an upload fails
+    DEFAULT_RETRY_TIME = 300
+    # interval time to wait between retry, in seconds
+    RETRY_INTERVAL_TIME = 0
+
+    def __init__(self, retry_time: int | None = None, **kwargs):
+        """
+        :param retry_time: maximum retry time, in seconds, when an upload fails
+        """
+        super().__init__(**kwargs)
+        self.retry_time = retry_time or self.DEFAULT_RETRY_TIME
```

Comment:
I'm pretty sure the mechanism for exponential backoff overriden by the server 429 header is already implemented in b2-sdk-python